### PR TITLE
Use -Wall in localization and local planner

### PR DESCRIPTION
**Pull request by timonegk**
## Proposed changes
Same as bit-bots/bitbots_motion#256.
Must be merged after bit-bots/particle_filter#9.

## Necessary checks
- [ ] Update package version
- [x] Run `catkin build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot
- [x] Put the PR on our Project board

### CODE REVIEW

File: bitbots_localization/src/RobotState.cpp [hunk 1]
```diff
@@ -8,17 +8,16 @@ RobotState::RobotState() :
     m_XPos(0.0),
     m_YPos(0.0),
     m_SinTheta(0.0),
-    m_CosTheta(1.0),
-    is_explorer_(false) {
+    m_CosTheta(1.0) {
```
Comment (Flova) [inline bitbots_localization/src/RobotState.cpp:11]: Why did you remove `is_explorer_(false)`?

Comment (timonegk) [reply bitbots_localization/src/RobotState.cpp]: By accident, thanks for spotting it.

**Review by Flova — APPROVED** [review]
